### PR TITLE
Update version to 3.2.0

### PR DESCRIPTION
**Pull request by ngranda**
This PR:
- bumps the dagmc version
- creates a changelog file with the consolidated news file

### CODE REVIEW

File: news/PR-0658.rst
Comment (ljacobson64) [inline news/PR-0658.rst]: Not "None"

Just an FYI but there needs to be a blank line between the `**Changed:**` line and the list of changes or else the list won't get formatted correctly

File: doc/CHANGELOG.rst
Comment (ljacobson64) [inline doc/CHANGELOG.rst]: Can this line be changed to "DAGMC Changelog"?

File: news/PR-0658.rst
Comment (ljacobson64) [reply news/PR-0658.rst]: See here https://github.com/ngranda/DAGMC/blob/update_version/news/PR-0658.rst

**Comment by pshriwise** [comment]
Just curious as to where we're at with @makeclean's suggestion about building the following DAGMC executables:

Legacy:
- [ ] FluDAG
- [ ] ~~DAG-Geant4~~
- [x] DAG-MCNP5
- [x] DAG-MCNP6
- [x] DAG-OpenMC

UWUW:
- [ ] FluDAG
- [x] DAG-Geant4
- [x] DAG-MCNP5
- [x] DAG-MCNP6
- [x] DAG-OpenMC

Figure we can check these off here if people have built them recently. I can account for DAG-MCNP6 with UWUW. And both workflows for OpenMC.

FWIW I know that DAG-GEANT4 is built with CI and it looks like the unit tests are run as well. Might be worth running a simple simulation just to check it though.

Thanks for making this happen @ngranda!

**Comment by ljacobson64** [comment]
As far as I'm aware, there is no distinction between "legacy" and "UWUW" at the executable level. All executables are supposed to be able to handle both. I can confirm that DAG-MCNP5, DAG-MCNP6, FluDAG, and Dag-Geant4 all compile correctly with the latest version of the repo. Where does this leave us?

**Comment by pshriwise** [comment]
@ljacobson64 I think what @makeclean was recommending (on slack) is that we check that the executables both compile and actually work with non-UWUW and UWUW. In principle, the executables shouldn't care how materials are defined and it's likely they work fine, but it might be worth running them with simple problems once - as long as it isn't too much of a hassle between all of us to do so. If nothing else, it'd just be nice to have this release PR as a reference that they all work at this point in time. If we feel like it's going to needlessly delay this release, then maybe it isn't worth it.

**Comment by ljacobson64** [comment]
@pshriwise That makes sense. I think this is a good idea. The question will be where will we get sample problems for all these cases?

**Comment by gonuke** [comment]
As implied by @pshriwise, it would be great if this kind of testing of a release candidate was a group effort.  Different parties have different levels of experience/expertise with each physics package and it would be easiest for each to test within their own expertise.

**Comment by ljacobson64** [comment]
I can't get DAG-MCNP5 or DAG-MCNP6 to work with legacy material assignments. I assigned the group name `mat:1/rho_-1.0` to a volume with all other volumes being `mat:Vacuum` or `mat:Graveyard` and this is what I get:

```
 mcnp     ver=6.mpi, ld=12/27/19  01/16/20 16:53:35                   
Loading file test.h5m
Initializing the GeomQueryTool...
Using faceting tolerance: 0.001
Building OBB Tree...
Using the simplified nameing scheme without a density
property is forbidden, please rename the group mat:1
```

**Comment by makeclean** [comment]
Its density rather than rho as a keyword right?
________________________________

**Comment by ljacobson64** [comment]
False alarm. I made a typo. I entered `mat:1/rho_-1.0` instead of `mat:1/rho:-1.0`.

**Comment by pshriwise** [comment]
Thanks for verifying MCNP5/6 w/ legacy @ljacobson64! So we're just looking for Legacy/UWUW Fluka & Geant4 at this point, right?

**Comment by bam241** [comment]
@gonuke @pshriwise @makeclean @ljacobson64 @ngranda what is the status here ?

@pshriwise you mention the `DAG-G4` is built and test in the CI. Does it mean we can consider it as working ?

Anyone with experience about fluDAG ?

**Comment by makeclean** [comment]
Im trying to get to find the time to get the new O/S version and test it, but im pretty busy. If any one can find the time - I can offer help trouble shooting



On 4 Feb 2020, at 14:59, Baptiste Mouginot <notifications@github.com<mailto:notifications@github.com>> wrote:


@gonuke<https://github.com/gonuke> @pshriwise<https://github.com/pshriwise> @makeclean<https://github.com/makeclean> @ljacobson64<https://github.com/ljacobson64> @ngranda<https://github.com/ngranda> what is the status here ?

@pshriwise<https://github.com/pshriwise> you mention the DAG-G4 is built and test in the CI. Does it mean we can consider it as working ?

Anyone with experience about fluDAG ?

**Comment by ljacobson64** [comment]
FluDAG builds without errors and doesn't immediately segfault when you run the executable with no input file. And the FluDAG unit tests do NOT pass. That's all I can say about FluDAG.

**Comment by pshriwise** [comment]
@bam241 thanks for nudging this. I noted that we consistently build DAG-GEANT4 in CI, but we should probably test it like all the others. I'll try to carve out some time to put together a couple of simple demo problems (legacy & UWUW) in the next week and try them out - unless someoone else has a better recommendataion for testing this.

**Comment by makeclean** [comment]
Only that there is no legacy mode for geant4

On 4 Feb 2020, at 17:44, Patrick Shriwise <notifications@github.com<mailto:notifications@github.com>> wrote:


@bam241<https://github.com/bam241> thanks for nudging this. I noted that we consistently build DAG-GEANT4 in CI, but we should probably test it like all the others. I'll try to carve out some time to put together a couple of simple demo problems (legacy & UWUW) in the next week and try them out - unless someoone else has a better recommendataion for testing this.

**Comment by pshriwise** [comment]
Fair enough. Won't waste my time looking for that then!

**Comment by pshriwise** [comment]
DAG-Geant4 confirmed working!

**Comment by bam241** [comment]
> DAG-Geant4 confirmed working!

you're my Hero !

**Comment by bam241** [comment]
@gonuke @ngranda @ljacobson64 @pshriwise @kkiesling @makeclean 
anything holding this ?

**Comment by ljacobson64** [comment]
This PR should probably be rebased against `develop` before merging. Also, all the PR notes in the `news` folder from the PRs that have been merged since the time this PR was originally made should be added to the `CHANGELOG`.

**Comment by gonuke** [comment]
Are we waiting for confirmation that FluDAG still works?  Or is that done?

**Comment by bam241** [comment]
@ngranda could you rebase this against `svalinn/dagmc:3.2.0-rc2` ?

**Review by bam241 — APPROVED** [review]
LGTM!

**Review by pshriwise — APPROVED** [review]
Looks good to me. Thanks for putting this together @ngranda!

**Comment by bam241** [comment]
LGTM! merging.

**Pull request merged**
